audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/exception/ApplicationAsyncExceptionHandler.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.audit.exception;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.aop.interceptor.AsyncUncaughtExceptionHandler;
+
+import java.lang.reflect.Method;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A class that handles the Uncaught Exceptions thrown by the Async processes
+ */
+public class ApplicationAsyncExceptionHandler implements AsyncUncaughtExceptionHandler {
```

Comment:
Where is this being used?

Reply:
This is used in the `AsyncWebMvcConfiguration` class